src/prlib/prlib.cpp

```diff
@@ -37,20 +39,41 @@ PR_EXTERN void PrInitializeModule(sceGsZbuf zbuf)
     prRenderStuff.Initialize(zbuf);
 }
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrCleanupModule);
+PR_EXTERN void PrCleanupModule()
+{
+    prRenderStuff.Cleanup();
+    
+    PrCleanupModel(NULL);
+    PrCleanupAnimation(NULL);
+    PrCleanupCamera(NULL);
+    PrCleanupScene(NULL);
+    
+    prObjectDatabase.Cleanup();
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrInitializeScene);
+PR_EXTERN PR_SCENEHANDLE PrInitializeScene(sceGsDrawEnv1 *drawEnv, char *name, u_int fbp)
+{
+   return (PR_SCENEHANDLE)prObjectDatabase.CreateScene(drawEnv, name, fbp);
+}
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrInitializeSceneDBuff);
+PR_EXTERN void PrInitializeSceneDBuff(sceGsDBuff *buff, char *name, u_int fbp)
+{
+    PrSceneObject* scene = prObjectDatabase.CreateScene(&buff->draw0, name, fbp);
+    scene->m_pDBuff = buff;
+}
 
 INCLUDE_ASM(const s32, "prlib/prlib", PrCleanupScene);
 
-PR_EXTERN void PrSetSceneFrame(PR_SCENEHANDLE model, sceGsFrame frame)
+PR_EXTERN void PrSetSceneFrame(PR_SCENEHANDLE scene, sceGsFrame frame)
 {
     // Empty
 }
 
-INCLUDE_ASM(const s32, "prlib/prlib", PrSetSceneEnv);
+PR_EXTERN void PrSetSceneEnv(PR_SCENEHANDLE scene, sceGsDrawEnv1 *drawEnv)
```

Comment:
You can freely use the classes on the .cpp file and get rid of the cast.
Please replace every mention of the typedefs with their respective classes.

Reply:
This causes a conflict with the header file's definition.

Reply:
Hmm, perhaps the .cpp file can include another header, while making the functions at `prlib/prlib.h` externs, because I doubt they wrote it like this. It's a statically linked library after all.